tests/test_circulationapi.py

```diff
@@ -929,6 +931,45 @@ def yes_we_can(*args, **kwargs):
         result = try_to_fulfill()
         assert fulfillment == result
 
+    def test_fulfill_cached_manifest_axis_ebook(self):
+        """
+        GIVEN: 
+        WHEN:  
+        THEN:  
+        """
+        self.collection = MockAxis360API.mock_collection(self._db)
+        self.patron = self._patron()
+        _, self.pool = self._edition(
+            data_source_name=DataSource.AXIS_360,
+            identifier_id='0016820953',
+            identifier_type=Identifier.AXIS_360_ID,
+            with_license_pool=True
+        )
+        loan, _ = self.pool.loan_to(self.patron, end=datetime(2018, 9, 29, 18, 34))
+
+        def mock_axis_api(licensepool):
+            api = MockAxis360API(self._db, self.collection)
+            api.queue_response(
+                200, {}, sample_data("availability_with_axisnow_fulfillment.xml", "axis"))
+            api.queue_response(
+                200, {}, sample_data("ebook_fulfillment_info.json", "axis"))
+            return api
+
+        self.circulation.api_for_license_pool = mock_axis_api
```

Comment:
This needs to be split up into a few different tests, some in api/axis.py and some here in circulation.py.

For axis.py, you need to verify that an outgoing AxisNow manifest has `can_cache_manifest` set to True, but that Adobe and Findaway manifests have the field set to False. You should be able to do this by adding assertions to the existing tests, instead of writing new ones.

For the test of `CirculationAPI`, you don't need any AxisNow-specific details, because the only thing that matters is whether `can_cache_manifest` is true. 

* If there's already a Loan object with `.cached_manifest` set, then `fulfill` returns a `FulfillmentInfo` with the appropriate media type and content, without making any API requests at all.
* If not, then `fulfill` makes the API call and sets `Loan.cached_manifest` and `Loan.cached_content_type`, but only if `Fulfillment.can_cache_manifest` is true. For the first case you can queue the FulfillmentInfo object directly with `self.remote.queue_fulfillment` (lots of examples in test_circulationapi.py) and you don't have to set up a mock API. For the second case you should be able to add assertions to an existing test, since the default behavior is that `Fulfillment.can_cache_manifest` is False.